lsf/res/res.handler.c

```diff
@@ -2869,17 +2877,147 @@ lsbExecChild(struct resCmdBill *cmdmsg, int *pty,
            pid,
            iofd,
            -1,
-           (cmdmsg->options & REXF_USEPTY));
+           (cmdmsg->options & REXF_USEPTY),
+           0);
     _exit(127);
 }
 
+
+static int get_argc(const char *argv[])
+{
+    int ret=0;
+    while (argv[ret]) ++ret;
+    return ret;
+}
+
+
+static char *join(int sz, const char *argv[])
+{
+    if (sz<0) sz=get_argc(argv);
+    int *argl=malloc(sizeof(int)*sz);
+    if (!argl) goto clean;
+    int i,len=sz-1;
+    for (i=0;i<sz;++i) len+=(argl[i]=strlen(argv[i]));
+    char *ret = malloc(len+1);
+    if (!ret) goto clean;
+    char *p=ret;
+    for (i=0;i<sz;++i)
+    {
+        memcpy(p,argv[i],argl[i]);
+        p+=argl[i];
+        if (i==sz-1) *p=0; else *(p++)=' ';
+    }
+clean:
+    free(argl);
+    return ret;
+}
+
+
+static char *escape(const char *s)
+{
+    int i;
+    if (!s) return NULL;
+    int n=strlen(s);
+    char *ret=malloc(n*2+1);
+    if (!ret) return NULL;
+    for (i=0;i<n;++i)
+    {
+        ret[i*2]='\\';ret[i*2+1]=s[i];
+    }
+    ret[2*n]=0;
+    return ret;
+}
+
+
+static void exec_task_on_azure(const char *argv[])
+{
+    FILE *f=NULL;
+    char **words=NULL;
+    char *cmd_string=NULL;
+    char *cmd_string_esc=NULL;
+    const char **chunks=NULL;
+    char *cmd_call_python=NULL;
+    char *python_script_path=NULL;
+
+    int err=0,i;
+    int argc=get_argc(argv);
+    
+    words = calloc(argc, sizeof(char *));
+    if (!words) {err=1;goto clear;}
+    
+    for (i=0;i<argc;++i)
+    {
+        words[i]=escape(argv[i]);
+        if (!words[i]) {err=1;goto clear;}
+    }
+    
+    cmd_string = join(argc, (const char**)words);
+    if (!cmd_string) {err=1;goto clear;}
+
+    cmd_string_esc = escape(cmd_string);
```

Comment:
Not sure why these escapes needed.
Could you try comments here for sample data conversion, so it's better to understand.